src/directory/footer.rs

```diff
@@ -86,16 +89,41 @@ impl Footer {
 
     /// Confirms that the index will be read correctly by this version of tantivy
     /// Has to be called after `extract_footer` to make sure it's not accessing uninitialised memory
-    pub fn is_compatible(&self) -> bool {
-        let version = &*crate::VERSION;
+    pub fn is_compatible(&self) -> Result<bool, TantivyError> {
+        let library = &*crate::VERSION;
         match &self.versioned_footer {
             VersionedFooter::V1 {
                 crc32: _,
                 compression,
             } => {
-                return compression == &version.store_compression;
+                // V1 means should be equal with 1
+                if library.index_format_version != 1u32 {
+                    if &library.store_compression != compression {
+                        return Err(IncompatibleIndex(
+                            Incompatibility::CompressionAndIndexMismatch {
+                                library: library.clone(),
+                                index: self.version.clone(),
+                            },
+                        ));
+                    } else {
+                        return Err(IncompatibleIndex(Incompatibility::IndexMismatch {
+                            library: library.clone(),
+                            index: self.version.clone(),
+                        }));
+                    }
+                }
+
+                if &library.store_compression != compression {
+                    return Err(IncompatibleIndex(Incompatibility::CompressionMismatch {
+                        library: library.clone(),
+                        index: self.version.clone(),
+                    }));
+                }
+
+                Ok(true)
             }
-            VersionedFooter::UnknownVersion { version: _ } => false,
+            // TODO - discuss if we should add another error type
+            VersionedFooter::UnknownVersion { version: _ } => return Ok(false),
```

Comment:
and return an `Err(IncompatibleIndex)` here instead.

Reply:
If I am going to change the return type to 
`Result<(), Incompatibility>`, I cannot return `Err(IncompatibleIndex)` but I can return `Err(Incompatibility)`